### PR TITLE
Make VPN Network Range Configurable

### PR DESCRIPTION
**Pull request by SimonKienzler**
**How to categorize this PR?**

/area networking
/kind enhancement

**What this PR does / why we need it**:

This PR introduces a new field to the `Seed.spec.networks` section called `vpn`, which allows operators to configure the previously hard-coded VPN Network on a per-seed basis.

**Which issue(s) this PR fixes**:

Part of #8987 

**Special notes for your reviewer**:

Please advise if I missed any parts of the code base where this change needs to be addressed.

**Release note**:

```feature operator
VPN Network can now be configured on a per-seed basis.
```

### CODE REVIEW

File: pkg/apis/core/validation/seed.go [hunk 1]
```diff
@@ -270,6 +271,17 @@ func validateSeedNetworks(seedNetworks core.SeedNetworks, fldPath *field.Path, i
 	if seedNetworks.Nodes != nil {
 		networks = append(networks, cidrvalidation.NewCIDR(*seedNetworks.Nodes, fldPath.Child("nodes")))
 	}
+	if seedNetworks.VPN != nil {
+		vpnNetwork = *seedNetworks.VPN
+	} else {
+		if core.IsIPv6SingleStack(seedNetworks.IPFamilies) {
```
Comment (oliver-goetz) [inline pkg/apis/core/validation/seed.go:277]: Shouldn't the default IP range depend on `primaryIPFamily` instead?
That's at least what's checked a few lines below
`allErrs = append(allErrs, cidrvalidation.ValidateCIDRIPFamily(networks, string(primaryIPFamily))...)`

File: pkg/component/vpnseedserver/vpn_seed_server.go [hunk 2]
```diff
@@ -122,6 +122,8 @@ type NetworkValues struct {
 	ServiceCIDR string
 	// NodeCIDR is the CIDR of the node network.
 	NodeCIDR string
+	// VPNCIDR is the CIDR of the VPN network.
+	VPNCIDR string
```
Comment (oliver-goetz) [inline pkg/component/vpnseedserver/vpn_seed_server.go:126]: What's the scope of this property? It is not used anywhere.

File: pkg/component/vpnshoot/vpnshoot.go [hunk 3]
```diff
@@ -86,6 +86,8 @@ type ReversedVPNValues struct {
 	OpenVPNPort int32
 	// IPFamilies are the IPFamilies of the shoot.
 	IPFamilies []gardencorev1beta1.IPFamily
+	// Network is the network CIDR for the ReversedVPN.
+	Network string
```
Comment (oliver-goetz) [inline pkg/component/vpnshoot/vpnshoot.go:90]: This property is used neither.

**Review by oliver-goetz — CHANGES_REQUESTED** [review]
Thanks for the PR 🙂 

This PR does not look complete. You have added a property for the VPN CIDR to the API and to the vpnseedserver and vpnshoot components too. However, these properties are not used anywhere.

/ok-to-test

**Comment by SimonKienzler** [comment]
/remove-lifecycle stale

We'll hopefully find some time to work on this in the upcoming weeks.

**Comment by timebertt** [comment]
/remove-lifecycle rotten

@SimonKienzler what's the status of this PR? Should we hand this one over to a colleague?

**Comment by SimonKienzler** [comment]
A handover would be greatly appreciated from my side ❤️

**Comment by timebertt** [comment]
/close
Unfortunately, this PR is quite conflicted with changes in the master branch.
I tried rebasing the commits but the PR is missing some things here and there.
To make things easier to review and follow, I will open a new PR with the missing parts and nicely structured commits.

**Pull request closed without merging**